### ISSUE TITLE
[Enhancement] Parser for hciconfig

### ISSUE THREAD

**Issue by borpin** [kellyjonbrazil/jc#100]
Any chance of a parser for `hciconfig` command - I want to monitor Up/Down status :)

Most of it is fairly well formatted in K/V pairs (separated by a colon), except for the bit I really want 😆 

```
$ hciconfig -a
hci0:   Type: Primary  Bus: USB
        BD Address: 00:1A:7D:DA:71:13  ACL MTU: 310:10  SCO MTU: 64:8
        UP RUNNING
        RX bytes:13905869 acl:0 sco:0 events:393300 errors:0
        TX bytes:62629 acl:0 sco:0 commands:3893 errors:0
        Features: 0xff 0xff 0x8f 0xfe 0xdb 0xff 0x5b 0x87
        Packet type: DM1 DM3 DM5 DH1 DH3 DH5 HV1 HV2 HV3
        Link policy: RSWITCH HOLD SNIFF PARK
        Link mode: SLAVE ACCEPT
        Name: 'CSR8510 A10'
        Class: 0x000000
        Service Classes: Unspecified
        Device Class: Miscellaneous,
        HCI Version: 4.0 (0x6)  Revision: 0x22bb
        LMP Version: 4.0 (0x6)  Subversion: 0x22bb
        Manufacturer: Cambridge Silicon Radio (10)

hci1:   Type: Primary  Bus: USB
        BD Address: 00:1A:7D:DA:71:13  ACL MTU: 310:10  SCO MTU: 64:8
        DOWN
        RX bytes:4388363 acl:0 sco:0 events:122021 errors:0
        TX bytes:52350 acl:0 sco:0 commands:3480 errors:2
        Features: 0xff 0xff 0x8f 0xfe 0xdb 0xff 0x5b 0x87
        Packet type: DM1 DM3 DM5 DH1 DH3 DH5 HV1 HV2 HV3
        Link policy: RSWITCH HOLD SNIFF PARK
        Link mode: SLAVE ACCEPT
```

**Comment by kellyjonbrazil** [comment]
Hi - thanks for the parser request! This looks like it would be a good addition. I’ll take a closer look at the output and see about getting this in the next release.

**Comment by borpin** [comment]
Brilliant :)

**Comment by kellyjonbrazil** [comment]
Taking a first stab at a schema for this. Let me know if you see anything glaring with this and I'll modify. I'll start working to fit the output to this schema:
```
[
  {
    "device": "hci0",
    "type": "Primary",
    "bus": "USB",
    "bd_address": "00:50:56:E8:36:2A",
    "acl_mtu": 8192,
    "acl_mtu_packets": 128,
    "sco_mtu": 64,
    "sco_mtu_packets": 128,
    "state": [
      "UP",
      "RUNNING"
    ],
    "rx_bytes": 1307,
    "rx_acl": 0,
    "rx_sco": 0,
    "rx_events": 51,
    "rx_errors": 0,
    "tx_bytes": 1200,
    "tx_acl": 0,
    "tx_sco": 0,
    "tx_commands": 51,
    "tx_errors": 0,
    "features": [
      "0xff",
      "0xff",
      "0x8f",
      "0xfe",
      "0x83",
      "0xe1",
      "0x08",
      "0x80"
    ],
    "packet_type": [
      "DM1",
      "DM3",
      "DM5",
      "DH1",
      "DH3",
      "DH5",
      "HV1",
      "HV2",
      "HV3"
    ],
    "link_policy": [
      "RSWITCH",
      "HOLD",
      "SNIFF",
      "PARK"
    ],
    "link_mode": [
      "SLAVE",
      "ACCEPT"
    ],
    "name": "foobar",
    "class": "0x000000",
    "service_classes": [
      "Unspecified"
    ],
    "device_class": "Miscellaneous", 
    "hci_version": "2.1 (0x4)",
    "revision": "0x100",
    "lmp_version": "2.1 (0x4)",
    "subversion": "0x100",
    "manufacturer": "not assigned (6502)"
  }
]
```

**Comment by kellyjonbrazil** [comment]
@borpin,

I have an initial working version that I plan to include in the next release. In the meantime you can test it by copying this file:

https://github.com/kellyjonbrazil/jc/blob/dev/jc/parsers/hciconfig.py

to the `$HOME/.local/share/jc/jcparsers` directory. (name the file `hciconfig.py`)

Let me know if that works ok for you.

Thanks!

**Comment by borpin** [comment]
Hi, I am getting
```
$ jc -d hciconfig hci0
Traceback (most recent call last):
  File "/home/pi/.local/bin/jc", line 10, in <module>
    sys.exit(main())
  File "/home/pi/.local/lib/python3.7/site-packages/jc/cli.py", line 456, in main
    result = parser.parse(data, raw=raw, quiet=quiet)
  File "/home/pi/.local/share/jc/jcparsers/hciconfig.py", line 403, in parse
    device_object['type'] = line[2]
IndexError: list index out of range
```

```
$ hciconfig hci0
hci0:   Type: Primary  Bus: USB
        BD Address: 00:1A:7D:DA:71:13  ACL MTU: 310:10  SCO MTU: 64:8
        DOWN
        RX bytes:629490 acl:0 sco:0 events:16606 errors:0
        TX bytes:2417 acl:0 sco:0 commands:128 errors:0
```

**Comment by borpin** [comment]
>     "state": [
>       "UP",
>       "RUNNING"
>     ],

Can also be DOWN and "INIT RUNNING" (IIRC).

**Comment by kellyjonbrazil** [comment]
Thanks for testing! I was looking for spaces instead of tabs (classic!). I have fixed the parser to look for any whitespace at the beginning of each line now. Could you re-copy the new code from the link above and try again?

**Comment by borpin** [comment]
Yes that works great :)

**Comment by kellyjonbrazil** [comment]
Excellent - I'll write some tests and probably get it out in v1.14.3 in the next day or so.

**Comment by kellyjonbrazil** [comment]
The new `hciconfig` parser has been merged into v1.14.3, now available on pypi via pip.
